### ISSUE TITLE
Can't compile sluice on Fedora Rawhide

### ISSUE THREAD

**Issue by ignatenkobrain** [sagebind/sluice#10]
```rust
   Compiling sluice v0.5.1 (/home/brain/rpmbuild/BUILD/sluice-0.5.1)
     Running `/usr/bin/rustc --edition=2018 --crate-name sluice src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C metadata=03b9863647ea8ece -C extra-filename=-03b9863647ea8ece --out-dir /home/brain/rpmbuild/BUILD/sluice-0.5.1/target/release/deps -L dependency=/home/brain/rpmbuild/BUILD/sluice-0.5.1/target/release/deps --extern futures_channel=/home/brain/rpmbuild/BUILD/sluice-0.5.1/target/release/deps/libfutures_channel-2d4bc6c413b60832.rmeta --extern futures_core=/home/brain/rpmbuild/BUILD/sluice-0.5.1/target/release/deps/libfutures_core-cbe64fd3f840fa1d.rmeta --extern futures_io=/home/brain/rpmbuild/BUILD/sluice-0.5.1/target/release/deps/libfutures_io-6ef128f8917047b3.rmeta --extern futures_util=/home/brain/rpmbuild/BUILD/sluice-0.5.1/target/release/deps/libfutures_util-0c17dd2ad5e16723.rmeta -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now -Ccodegen-units=1 --cap-lints=warn`
error[E0432]: unresolved import `futures_util::io`
  --> src/pipe/chunked.rs:26:19
   |
26 | use futures_util::io::Cursor;
   |                   ^^ could not find `io` in `futures_util`

error[E0599]: no method named `poll_next` found for type `std::pin::Pin<_>` in the current scope
  --> src/pipe/chunked.rs:92:57
   |
92 |                 match Pin::new(&mut self.buf_stream_rx).poll_next(cx) {
   |                                                         ^^^^^^^^^ method not found in `std::pin::Pin<_>`
   |
   = note: Pin::new(&mut self.buf_stream_rx) is a function, perhaps you wish to call it

error[E0599]: no method named `poll_read` found for type `std::pin::Pin<&mut _>` in the current scope
   --> src/pipe/chunked.rs:106:46
    |
106 |         let len = match Pin::new(&mut chunk).poll_read(cx, buf) {
    |                                              ^^^^^^^^^ method not found in `std::pin::Pin<&mut _>`
    |
    = note: Pin::new(&mut chunk) is a function, perhaps you wish to call it

error[E0599]: no method named `poll_next` found for type `std::pin::Pin<_>` in the current scope
   --> src/pipe/chunked.rs:170:47
    |
170 |         match Pin::new(&mut self.buf_pool_rx).poll_next(cx) {
    |                                               ^^^^^^^^^ method not found in `std::pin::Pin<_>`
    |
    = note: Pin::new(&mut self.buf_pool_rx) is a function, perhaps you wish to call it

warning: unused import: `FusedStream`
  --> src/pipe/chunked.rs:24:20
   |
24 | use futures_core::{FusedStream, Stream};
   |                    ^^^^^^^^^^^
   |
note: lint level defined here
  --> src/lib.rs:18:5
   |
18 |     unused,
   |     ^^^^^^
   = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

warning: unused import: `Stream`
  --> src/pipe/chunked.rs:24:33
   |
24 | use futures_core::{FusedStream, Stream};
   |                                 ^^^^^^

error: aborting due to 4 previous errors
```

**Comment by sagebind** [comment]
Huh, this seems like really weird Cargo behavior... cloning the repo and doing `cargo build` works just fine. Even my CI passes! But if I comment out `futures` from `[dev-dependencies]`, this error occurs because the `io` feature was no longer getting enabled by the dev-dependency...

Thanks for catching this though!

**Comment by ignatenkobrain** [comment]
The fix in #11 . You can reproduce this with passing -Zavoid-dev-deps... Which hopefully will be stabilized at some point. It catches bugs like this from time to time.

**Comment by sagebind** [comment]
Looks like this is an accidental regression from #7.

**Comment by sagebind** [comment]
The fix will go out in 0.5.2, which will be ready this week.
